### PR TITLE
Gas price config

### PR DESCRIPTION
**Pull request by LogvinovLeon**
This PR:
* Allows the user to specify gas price for transactions when instantiating `ZeroEx`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # CHANGELOG
 
+vTBD - _TBD_
+    * Added a possibility to specify `gasPrice` when instantiating `ZeroEx` (#139)
```
Comment (fabioberger) [inline CHANGELOG.md]: `Added the ability to specify `gasPrice` when instantiating `ZeroEx` (#139)`

File: src/0x.ts [hunk 2]
```diff
@@ -162,15 +162,16 @@ export class ZeroEx {
      * Instantiates a new ZeroEx instance that provides the public interface to the 0x.js library.
      * @param   provider    The Web3.js Provider instance you would like the 0x.js library to use for interacting with
      *                      the Ethereum network.
+     * @param   gasPrice    The gas price to use for sending transactions. Defaults to 21 GWei.
      * @return  An instance of the 0x.js ZeroEx class.
      */
-    constructor(provider: Web3Provider) {
+    constructor(provider: Web3Provider, gasPrice?: BigNumber.BigNumber) {
```
Comment (fabioberger) [inline src/0x.ts]: So I really do feel strongly that we should make this a config obj called `configs`. I am 100% sure this won't be the last config param we will add to this library and it'll be cleaner to write:

```
const configs = {
  gasPrice,
  requestTimeoutLimit,
  verbose,
};
const zeroEx = new ZeroEx(provider, configs);
```

Then:

```
const zeroEx = new ZeroEx(provider, gasPrice, requestTimeoutLimit, verbose);
```

Especially as the number of configs grows.

File: src/0x.ts [hunk 3]
```diff
@@ -162,15 +162,16 @@ export class ZeroEx {
...
         this._web3Wrapper = new Web3Wrapper(provider);
-        this.token = new TokenWrapper(this._web3Wrapper);
-        this.proxy = new TokenTransferProxyWrapper(this._web3Wrapper);
-        this.exchange = new ExchangeWrapper(this._web3Wrapper, this.token);
-        this.tokenRegistry = new TokenRegistryWrapper(this._web3Wrapper);
-        this.etherToken = new EtherTokenWrapper(this._web3Wrapper, this.token);
+        this.token = new TokenWrapper(this._web3Wrapper, gasPrice);
```
Comment (fabioberger) [inline src/0x.ts]: The above change will also keep all of these sub-classes cleaner too!

Comment (LogvinovLeon) [reply src/0x.ts]: I'll leave those subclasses alone for now, cause they're private and it's easy to change their interface later.

Comment (fabioberger) [reply src/0x.ts]: @LogvinovLeon if you don't do it now, you will do it once you add another param. Are you sure you don't want to take care of it now?

**Comment by LogvinovLeon** [comment]
Why would we want to pass the whole library config object down to specific classes?
It's like passing the whole redux state to pure components.
Later - it there is enough configs - we can rethink it.

**Pull request merged**